### PR TITLE
Allow social & search definitions to be sourced locally

### PR DESCRIPTION
**Pull request by samjf**
### Description:

Issue: DEV-2088

**Introduced functionality:**
This change allows Matomo to source [search + social definitions yaml files](https://github.com/matomo-org/searchengine-and-social-list) locally, instead of reading them from remotely sourced updates.

If the introduced configuration variable is changed to 0 (false) then it will only ever load social and search definitions from the local YML files packaged with Matomo. 

 **Existing behavior**
How it works at the moment is that a weekly task will fetch the latest definition files from [Matomo github](https://github.com/matomo-org/searchengine-and-social-list). It will store this in the the Option table and retrieve it from the option table when definitions are requested and there is a cache miss.

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: config/global.ini.php [hunk 1]
```diff
@@ -833,6 +833,10 @@
 ; If set to 1, when rearchiving reports in the past we do not rearchive segment data with those reports. Default is 0.
 rearchive_reports_in_past_exclude_segments = 0
 
+; If set to 1, then social and search engine definitions files will be downloaded from Matomo remotely to update
```
Comment (tsteur) [inline config/global.ini.php]: These are actually downloaded from github see https://github.com/matomo-org/matomo/blob/4.4.1/plugins/Referrers/Tasks.php#L33-L49 and not from Matomo. This is something we need to improve in the future to "proxy" or serve them through Matomo servers.

**Review by tsteur — COMMENTED** [review]
Left a quick comment @samjf I haven't tested yet but code looks good. Still needs to be tested (by me or some other reviewer)

File: config/global.ini.php (hunk 1 above)
Comment (samjf) [reply config/global.ini.php]: @tsteur Thanks for that. I've made a change based on your suggestion :)

File: config/global.ini.php [hunk 2]
```diff
@@ -835,6 +835,10 @@
 ; If set to 1, when rearchiving reports in the past we do not rearchive segment data with those reports. Default is 0.
 rearchive_reports_in_past_exclude_segments = 0
 
+; If set to 1, then social and search engine definitions files will be will be synchronised using the internet.
```
Comment (sgiehl) [inline config/global.ini.php]:
```suggestion
; If set to 1, then social and search engine definitions files will be synchronised using the internet.
```

File: config/global.ini.php [hunk 3]
```diff
@@ -835,6 +835,10 @@
...
+; When set to 0 the definitions will be loaded from the local definitions (updated with core).
+enable_referrer_definition_syncs = 1
```
Comment (sgiehl) [inline config/global.ini.php:840]: Should we mention here that this setting actually requires `enable_internet_features` to be enabled?

Comment (tsteur) [reply config/global.ini.php:840]: Can mention it 👍

**Pull request merged**